tracer/src/Datadog.Trace/AppSec/Waf/Initialization/LibraryLoader.cs

```diff
@@ -235,7 +235,17 @@ private static void GetLibNameAndRuntimeId(FrameworkDescription frameworkDescrip
             if (runtimeIdPart1 != null && libPrefix != null && libExt != null)
             {
                 libName = libPrefix + "ddwaf." + libExt;
-                runtimeId = Environment.Is64BitProcess ? runtimeIdPart1 + "-x64" : runtimeIdPart1 + "-x86";
+                string procArch;
+                if (frameworkDescription.ProcessArchitecture == ProcessArchitecture.Arm64)
+                {
+                    procArch = ProcessArchitecture.Arm64;
+                }
+                else
+                {
+                    procArch = Environment.Is64BitProcess ? ProcessArchitecture.X64 : ProcessArchitecture.X86;
+                }
```

Comment:
nit; I think this _might_ be easier to follow as a switch? (wish we could use value tuples to make this even clearer 🙁)

```suggestion
                string procArch = frameworkDescription.ProcessArchitecture switch
                {
                    ProcessArchitecture.Arm64 => ProcessArchitecture.Arm64,
                    _ when Environment.Is64BitProcess => ProcessArchitecture.X64,
                    _ => ProcessArchitecture.X86,
                };
```

Reply:
Yes, you're right, expressions are nearly always a better choice than statements.